tests/operators/test_operator_norm.py

```diff
@@ -0,0 +1,98 @@
+"""Tests for computing the operator norm for linear operators."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from math import prod
+from math import sqrt
+
+import torch
+from mrpro.operators import EinsumOp
+from mrpro.operators import FastFourierOp
+
+from tests import RandomGenerator
+
+
+def test_operator_norm_result():
+    """Test if the implementation yields the correct result for different choices
+    of operators with known operator-norm."""
+    random_generator = RandomGenerator(seed=0)
+
+    # test with a 3x3 matrix with known largest eigenvalue
+    matrix1 = torch.tensor([[2.0, 1, 0], [1.0, 2.0, 1.0], [0.0, 1.0, 2.0]])
+    operator1 = EinsumOp(matrix1, 'y x, x-> y')
+    random_vector1 = random_generator.float32_tensor(matrix1.shape[1])
+    operator1_norm_est = operator1.operator_norm(random_vector1, dim=None, max_iterations=32)
+    operator1_norm_true = 2 + sqrt(2)  # approximately 3.41421...
+    torch.testing.assert_close(operator1_norm_est.item(), operator1_norm_true, atol=1e-4, rtol=1e-4)
+
+
+def test_fourier_operator_norm():
+    """Test with Fast Fourier Operator (has norm 1 since norm="ortho" is used);
+    # also tests that the initial value is set to a random value if chosen as zero."""
```

Comment:
Would it make sense to have a test which ensures that the same norm value is calculated independent if you give a random value oor chose the input as 0?

Reply:
i think it's so trivial that probably not worthy to be tested. Also, the fact that the test passes although we explicitly set zero as the initial value covers the case imho.